easybuild/easyblocks/m/maple.py

```diff
@@ -35,61 +35,75 @@
+        cmd = os.path.join(self.builddir, self.src[0]['name'])
 
-        qa = {
-              'PRESS <ENTER> TO CONTINUE:': '',
-              'DO YOU ACCEPT THE TERMS OF THIS LICENSE AGREEMENT? (Y/N):': 'Y',
-              'ENTER AN ABSOLUTE PATH, OR PRESS <ENTER> TO ACCEPT THE DEFAULT :': self.installdir,
-              'IS THIS CORRECT? (Y/N):': 'Y',
-              'Do you wish to have a shortcut installed on your desktop? ->1- Yes 2- No ENTER THE NUMBER FOR YOUR CHOICE, OR PRESS <ENTER> TO ACCEPT THE DEFAULT::': '2',
-              '->1- Single User License 2- Network License ENTER THE NUMBER FOR YOUR CHOICE, OR PRESS <ENTER> TO ACCEPT THE DEFAULT::': '2',
-              'PRESS <ENTER> TO EXIT THE INSTALLER:': '',
-              'License server (DEFAULT: ):': self.cfg['license_server'],
-              'Port number (optional) (DEFAULT: ):': '',
-              '->1- Configure toolbox for Matlab 2- Do not configure at this time ENTER THE NUMBER FOR YOUR CHOICE, OR PRESS <ENTER> TO ACCEPT THE DEFAULT::': '2'
-             }
+        license_option = None
+        if self.cfg['license_server'] is not None:
+            # Network License
+            license_option = '2'
+        else:
+            self.log.error("A license server should be specified.")
 
-        no_qa = ['Graphical installers are not supported by the VM. The console mode will be used instead...',
-                 'Extracting the JRE from the installer archive...',
-                 'Launching installer...',
-                 "Configuring the installer for this system's environment...",
-                 'Unpacking the JRE...',
-                 '\[[-|]*']
-
-        run_cmd_qa(cmd, qa, no_qa=no_qa, log_all=True, simple=True)
+        choose_number = "ENTER THE NUMBER FOR YOUR CHOICE, OR PRESS <ENTER> TO ACCEPT THE DEFAULT::"
+        qa = {
+            "PRESS <ENTER> TO CONTINUE:": '',  # Maple 15, 17
+            "Press [Enter] to continue:": '',  # Maple 18
+            "DO YOU ACCEPT THE TERMS OF THIS LICENSE AGREEMENT? (Y/N):": 'Y',  # Maple 15, 17
+            "Do you accept this license? [y/n]:": 'y',  # Maple 18
+            "ENTER AN ABSOLUTE PATH, OR PRESS <ENTER> TO ACCEPT THE DEFAULT :": self.installdir,  # Maple 15, 17
+            "IS THIS CORRECT? (Y/N):": 'Y',
+            "Do you wish to have a shortcut installed on your desktop? ->1- Yes 2- No " + choose_number: '2',  # Maple 15, 17
+            "->1- Single User License 2- Network License " + choose_number: license_option,
+            "PRESS <ENTER> TO EXIT THE INSTALLER:": '',
+            "Port number (optional) (DEFAULT: ):": '',
+            "->1- Configure toolbox for Matlab 2- Do not configure at this time " + choose_number: '2',  # Maple 15
+            "->1- Configure toolbox for MATLAB 2- Do not configure at this time " + choose_number: '2',  # Maple 17
+            "MATLAB Configuration [y/N]:": 'n',  # Maple 18
```

Comment:
don't get me started on an ideal world... ;-)

This can be figured out when people have a need for it.